### PR TITLE
Minor improvements to C++ support, including support on stm32

### PR DESCRIPTION
**Pull request by dpgeorge**
Changes include:
- support C++ in stm32 port (still WIP, only works on boards using common_ifs.ld)
- build user C modules, including C++ module, in stm32 CI job
- add link to user C code on GitHub in user C modules docs
- only build coverage.c and coveragecpp.cpp as part of unix coverage variant, to remove need for C++ compiler in non-coverage unix builds
- fix extreme_exc.py test that fails locally for me on coverage build with coveragecpp.cpp included

### CODE REVIEW

**Comment by dpgeorge** [comment]
@stinos FYI

File: ports/stm32/Makefile [hunk 1]
```diff
@@ -138,6 +138,14 @@ else
 COPT += -Os -DNDEBUG
 endif
 
+# Flags for optional C++ source code
+CXXFLAGS += $(filter-out -Wmissing-prototypes -Wold-style-definition -std=gnu99,$(CFLAGS))
+CXXFLAGS += $(CXXFLAGS_MOD)
+ifneq ($(SRC_USERMOD_CXX),)
+$(info heeer)
```
Comment (stinos) [inline ports/stm32/Makefile:145]: debug leftover?

**Comment by stinos** [comment]
Looks ok!

File: ports/stm32/Makefile (hunk 1 above)
Comment (dpgeorge) [reply ports/stm32/Makefile]: yep, now fixed

**Pull request merged**